uavcan/equipment/ice/reciprocating/1120.Status.uavcan

```diff
@@ -0,0 +1,178 @@
+# Next states: STARTING, FAULT
+#
+uint2 STATE_STOPPED = 0
+#
+# The engine is starting. This is a transient state.
+# Next states: STOPPED, RUNNING, FAULT
+#
+uint2 STATE_STARTING = 1
+#
+# The engine is running normally.
+# Some error flags may be set to indicate non-fatal issues, e.g. overheating.
+# Next states: STOPPED, FAULT
+#
+uint2 STATE_RUNNING = 2
+#
+# The engine can no longer function.
+# The error flags may contain additional information about the nature of the fault.
+# Next states: STOPPED.
+#
+uint2 STATE_FAULT = 3
+
+#
+# General status flags.
+# Note that not all flags are required. Those that aren't are prepended with a validity flag, which is, obviously,
+# always required; when the validity flag is set, it is assumed that the relevant flags are set correctly.
+# If the validity flag is cleared, then the state of the relevant flags should be ignored.
+# All unused bits must be cleared.
+#
+uint22 flags
+#
+# General error. This flag is required, and it can be used to indicate an error condition
+# that does not fit any of the other flags.
+# Note that the vendor may also report additional status information via the vendor specific status code
+# field of the NodeStatus message.
+#
+uint22 FLAG_GENERAL_ERROR                       = 1
+#
+# Error of the crankshaft sensor. This flag is optional.
+#
+uint22 FLAG_CRANKSHAFT_SENSOR_ERROR_SUPPORTED   = 2
+uint22 FLAG_CRANKSHAFT_SENSOR_ERROR             = 4
+#
+# Temperature levels. These flags are optional; either none of them or all of them are supported.
+#
+uint22 FLAG_TEMPERATURE_SUPPORTED               = 8
+uint22 FLAG_TEMPERATURE_BELOW_NOMINAL           = 16    # Under-temperature warning
+uint22 FLAG_TEMPERATURE_ABOVE_NOMINAL           = 32    # Over-temperature warning
+uint22 FLAG_TEMPERATURE_OVERHEATING             = 64    # Critical overheating
+uint22 FLAG_TEMPERATURE_EGT_ABOVE_NOMINAL       = 128    # Exhaust gas over-temperature warning 
+#
+# Fuel pressure. These flags are optional; either none of them or all of them are supported.
+#
+uint22 FLAG_FUEL_PRESSURE_SUPPORTED             = 256
+uint22 FLAG_FUEL_PRESSURE_BELOW_NOMINAL         = 512   # Under-pressure warning
+uint22 FLAG_FUEL_PRESSURE_ABOVE_NOMINAL         = 1024  # Over-pressure warning
+#
+# Detonation and misfire. These flags are optional; either none of them or all of them are supported.
+#
+uint22 FLAG_DETONATION_SUPPORTED                = 2048
+uint22 FLAG_DETONATION_OBSERVED                 = 4096  # Detonation condition observed warning
+uint22 FLAG_MISFIRE_OBSERVED                    = 8192  # Misfire condition observed warning
+
+#
+# Engine load estimate.
+# Unit: percent.
+# Range: [0, 127].
+#
+uint7 engine_load_percent
+
+#
+# Engine speed.
+# Unit: revolutions per minute.
+#
+uint17 engine_speed_rpm
+
+#
+# Estimated lambda coefficient.
+# This parameter is mostly useful for monitoring and tuning purposes.
+# Unit: dimensionless ratio
+#
+float16 lambda_coefficient
```

Comment:
I am not sure how the lambda is installed on planes at all. I have seen it only few times in my life and that was per-cylinder based. Usually there is no such sensor at all on real planes.
So this might be a question to discuss.
I am not sure that moving lambda to cylinders is right, however I am very sure that we can have several exhaust pipes from different cylinders and one sensor can cover only few of the available cylinders.
The question is what we have lambda for. If it is simply for average monitoring - it's fine to leave as it is, but if it is to check for fuel burn conditions and efficiency - should be more than one instance.
I prefer to have ECU manufacturers to comment on this.